### ISSUE TITLE
Implement Content Finder 

### ISSUE THREAD

**Issue by Minoost** [SapphireServer/Sapphire#40]
#### Sapphire
- [ ] Add ContentFinder class (server-wide global variable)
- [ ] Implement player queue
- [ ] Generate an instance as needed? (Discussion needed)
- [ ] Restore previous zone position where left off when he leaves from the duty

#### Exd
- [ ] Figure out how alliance raid is handled in ContentMemberType

#### Misc.
- [ ] Document CF packets

- More TBD

### Considerations/Notes
- It has a potential to share some codes with quest instance. (Which, by the way, is required to implement it around around lv5-6 MSQ)
- Opcode 0x78 is more like a union. We have to figure out how to handle this elegantly.

**Comment by Minoost** [comment]
Reserved

**Comment by Minoost** [comment]
```
Mordred - Today at 4:00 PM
Here is how i think it could work.
we'd define a class ContentFinder
and make that available as global object.
g_contentFinder
that will have a playerQueue, instanceTemplate, and activeInstance list.
playerQueue will be a certain structure with contentType, contentID ( if not a roulette ), role, job
for matchmaking.
when you have the needed players, a new instance is created from a template and moved to the activeList.
It will be assinged an internal id i guess
this id will have to be stored with the player
in the g_gameserver updateloop we'd trigger g_contentFinder.update() to perform logic
running zones, updating queue, etc.
btw, a player has two sets of positions in the DB
one is the zone/pos he entered an instance from, the 2nd is position inside the instance so that when he leaves he can be put back to where he was.
Think all that makes sense?
```

**Comment by Minoost** [comment]
Implemented penalty time in #55
